packages/web-components/fast-foundation/src/button/button.ts

```diff
@@ -157,6 +157,25 @@ export class Button extends FormAssociatedButton {
 
         this.proxy.setAttribute("type", this.type);
         this.handleUnsupportedDelegatesFocus();
+        document.addEventListener('click', this.handleClick, true)
```

Comment:
I don't think we need to watch the entire document to catch this - we could probably just listen on the internal `<button>` or `<span>` elements. That would provide ample opportunity to stop propagation without running this code for *every* click event that bubbles up to the document.

Reply:
Makes sense, something interesting to note here is that it doesn't seem possible to addEventLister to disabled buttons  , including the fast-button.
Might have something to do with the specs defined here (4.10.18.5): https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fe-disabled 

So I'm going to add it to the #shadowRoot > button > span instead.

Reply:
Yea that makes sense, thanks @prudepixie!